tests: improve Code Coverage of ConfigurationService (SDKCF-6388)

ts-robih-robihamanto opened the pull request:
# Description
Improved coverage of `ConfigurationService` to **98.9%**
This else line cannot failed, because it's already checked in the previous line if it's link is a valid URL:
```swift
guard let url = urlComponents?.url else {
    return .failure(RequestError.urlIsNil)
}
```

## Links
SDKCF-6388

# Checklist
- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have added to the [changelog](../blob/master/CHANGELOG.md#Unreleased)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys, and internal links
- [ ] I ran `fastlane ci` without errors
- [x] All project file changes are replicated in `SampleSPM/SampleSPM.xcodeproj` project

ts-robih-robihamanto committed:
tests: Add testing code to increase coverage in ConfigurationService (SDKCF-6388)

ts-robih-robihamanto committed:
chore: Update changelog to include current working ticket (SDKCF-6388)

imaginaris commented on Tests/Tests/ConfigurationServiceSpec.swift:
```diff
@@ -254,6 +254,15 @@ class ConfigurationServiceSpec: QuickSpec {
 
+            context("when building request body") {
+
+                it("will return RequestError.bodyIsNil") {
```
please add test cases that check if the same is returned when you pass a non-nil parameter in `buildHttpBody()` so it's clear that `bodyIsNil` is always returned no matter what parameter is passed

imaginaris reviewed (changes requested):
The `assertionFailure()` in `buildRequestHeader()` can be tested by setting subscription id to nil and using `to(throwAssertion())` predicate. It won't crash the unit test run

ts-robih-robihamanto committed:
tests: Add testing to cover more case  (SDKCF-6388)

Added for assertionFailure in building buildRequestHeader() and buildHttpBody()

ts-robih-robihamanto commented:
> The `assertionFailure()` in `buildRequestHeader()` can be tested by setting subscription id to nil and using `to(throwAssertion())` predicate. It won't crash the unit test run

I added the case for throwAssertion here [0726cfa](https://github.com/rakutentech/ios-inappmessaging/pull/261/commits/62da5bc73d4b56f83df4c5bd4f4234dd20726cfa).

imaginaris reviewed (approved):
👍
